libs/ui-lib/lib/common/components/ui/formik/InputField.tsx

```diff
@@ -43,57 +43,55 @@ const InputField: React.FC<
 
     const fieldId = getFieldId(props.name, 'input', idPostfix);
     const errorMessage = useFieldErrorMsg({ name: props.name, inputError: error, validate });
-    const fieldHelperText = <HelperText fieldId={fieldId}>{helperText}</HelperText>;
     const isValid = !(touched && error);
 
     return (
-      <Stack id={`form-control__${fieldId}`}>
-        <StackItem>
-          <FormGroup
-            fieldId={fieldId}
-            label={label}
-            helperText={fieldHelperText}
-            helperTextInvalid={fieldHelperText}
-            validated={isValid ? 'default' : 'error'}
-            isRequired={isRequired}
-            labelIcon={labelIcon}
-            labelInfo={labelInfo}
-          >
-            {description && (
-              <HelperText fieldId={fieldId}>
-                <HelperTextItem variant="indeterminate">{description}</HelperTextItem>
-              </HelperText>
-            )}
-            <Split>
-              <SplitItem isFilled>
-                <TextInput
-                  {...field}
-                  {...props}
-                  ref={ref}
-                  id={fieldId}
-                  validated={isValid ? 'default' : 'error'}
-                  isRequired={isRequired}
-                  aria-describedby={`${fieldId}-helper`}
-                  onChange={(value, event) => {
-                    if (!props.isDisabled) {
-                      !noDefaultOnChange && field.onChange(event);
-                      onChange && onChange(event);
-                    }
-                  }}
-                />
-              </SplitItem>
-              <SplitItem>{children}</SplitItem>
-            </Split>
-          </FormGroup>
-        </StackItem>
-        <StackItem>
-          {showErrorMessage && !isValid && (
-            <HelperText fieldId={fieldId} isError>
-              {errorMessage}
+      <FormGroup
+        id={`form-control__${fieldId}`}
+        fieldId={fieldId}
+        label={label}
+        isRequired={isRequired}
+        labelIcon={labelIcon}
+        labelInfo={labelInfo}
+      >
+        {description && (
+          <HelperText>
+            <HelperTextItem variant="indeterminate">{description}</HelperTextItem>
+          </HelperText>
+        )}
+        <Split>
+          <SplitItem isFilled>
+            <TextInput
+              {...field}
+              {...props}
+              ref={ref}
+              id={fieldId}
+              validated={isValid ? 'default' : 'error'}
+              isRequired={isRequired}
+              aria-describedby={`${fieldId}-helper`}
+              onChange={(event) => {
+                if (!props.isDisabled) {
+                  !noDefaultOnChange && field.onChange(event);
+                  onChange && onChange(event);
+                }
+              }}
+            />
+          </SplitItem>
+          <SplitItem>{children}</SplitItem>
+        </Split>
+        {(showErrorMessage || helperText) && (
```

Comment:
Before, we'd check that if `showErrorMessage` was true, the field was invalid (meaning there would be an error message to show).

Now I think this could attempt to display the block when there is no error. I'd keep the condition that there is an error to show / the field is invalid.

So something like `(showErrorMessage && !isValid) || helperText`.

But also as before, if we have this pattern in multiple places, it would be better to extract it, to avoid having to duplicate code and fix issues in multiple places.

Reply:
https://issues.redhat.com/browse/MGMT-16680

Reply:
The part of controlling when to show the helper text hasn't been fixed.

If `showErrorMessage` is true, `helperText` is false, and there is no error, it would show something when it shouldn't